core/types/header_ext_test.go

```diff
@@ -0,0 +1,146 @@
+// (c) 2025, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package types
+
+import (
+	"encoding/hex"
+	"encoding/json"
+	"math/big"
+	"reflect"
+	"testing"
+
+	"github.com/ava-labs/libevm/common"
+	ethtypes "github.com/ava-labs/libevm/core/types"
+	"github.com/ava-labs/libevm/rlp"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestHeaderRLP(t *testing.T) {
+	t.Parallel()
+
+	got := testHeaderEncodings(t, rlp.EncodeToBytes, rlp.DecodeBytes)
+
+	// Golden data from original coreth implementation, before integration of
+	// libevm. WARNING: changing these values can break backwards compatibility
+	// with extreme consequences as block-hash calculation may break.
+	const (
+		wantHex     = "f90234a00100000000000000000000000000000000000000000000000000000000000000a00200000000000000000000000000000000000000000000000000000000000000940300000000000000000000000000000000000000a00400000000000000000000000000000000000000000000000000000000000000a00500000000000000000000000000000000000000000000000000000000000000a00600000000000000000000000000000000000000000000000000000000000000b901000700000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000008090a0b0c0da00e00000000000000000000000000000000000000000000000000000000000000880f00000000000000a015000000000000000000000000000000000000000000000000000000000000001016171213a01400000000000000000000000000000000000000000000000000000000000000"
+		wantHashHex = "7b63c1c763cc1527f12de7e5331b92d66ea201d1e6d7cd65a79bf8e3b99c104d"
+	)
+
+	assert.Equal(t, wantHex, hex.EncodeToString(got), "Header RLP")
+
+	header, _ := headerWithNonZeroFields()
+	gotHashHex := header.Hash().Hex()
+	assert.Equal(t, "0x"+wantHashHex, gotHashHex, "Header.Hash()")
+}
+
+func TestHeaderJSON(t *testing.T) {
+	t.Parallel()
+
+	testHeaderEncodings(t, json.Marshal, json.Unmarshal)
```

Comment:
nit: can we explicitly ignore the return value and mention we aren't testing the JSON vs a gold standard or similar?

Reply:
Should we test against a gold standard? 🤔

Reply:
Up to you, I don't think it is strictly necessary like RLP. But also it is not harmful.